src/charm.py

```diff
@@ -315,9 +348,66 @@ def hostname(self) -> str:
     def _on_list_receivers_action(self, event: ops.ActionEvent):
         res = {}
         for receiver in self._requested_receivers():
-            res[receiver.replace("_", "-")] = f"{self._ingress.url or self.tempo.url}/{receiver}"
+            res[
+                receiver.replace("_", "-")
+            ] = f"{self._ingress.external_host or self.tempo.url}/{receiver}"
         event.set_results(res)
 
+    @property
+    def _static_ingress_config(self) -> dict:
+        entry_points = {}
+        for protocol, port in self.tempo.all_ports.items():
+            sanitized_protocol = protocol.replace("_", "-")
+            entry_points[sanitized_protocol] = {"address": f":{port}"}
+
+        return {"entryPoints": entry_points}
+
+    @property
+    def _ingress_config(self) -> dict:
```

Comment:
Would it make sense to extract this logic into an ops-independent module?

Reply:
There's a similar section in grafana-k8s-operator: https://github.com/canonical/grafana-k8s-operator/blob/main/src/charm.py#L1325

I think that yes, it might make sense, but as a separate issue.